src/job.ts

```diff
@@ -1102,7 +1102,7 @@ export class Job {
             await Promise.any(Object.keys(imageInspect[0].Config.ExposedPorts).map((port) => {
                 if (!port.endsWith("/tcp")) return;
                 const portNum = parseInt(port.replace("/tcp", ""));
-                const spawnCmd = ["docker", "run", "--rm", `--name=gcl-wait-for-it-${this.jobId}-${serviceIndex}-${portNum}`, "--network", `gitlab-ci-local-${this.jobId}`, "willwill/wait-for-it", `${uniqueAlias}:${portNum}`, "-t", "30"];
+                const spawnCmd = [this.argv.containerExecutable, "run", "--rm", `--name=gcl-wait-for-it-${this.jobId}-${serviceIndex}-${portNum}`, "--network", `gitlab-ci-local-${this.jobId}`, "willwill/wait-for-it", `${uniqueAlias}:${portNum}`, "-t", "30"];
```

Comment:
```suggestion
                const spawnCmd = [this.argv.containerExecutable, "run", "--rm", `--name=gcl-wait-for-it-${this.jobId}-${serviceIndex}-${portNum}`, "--network", `gitlab-ci-local-${this.jobId}`, "docker.io/willwill/wait-for-it", `${uniqueAlias}:${portNum}`, "-t", "30"];
```

We need to adapt this to use the complete docker image URL here. Otherwise it will break:

```
[...]
[13:40:52    21 s] tests started service image: docker.io/some/image:tag with aliases: docker.io-some-image, docker.io__some__image, myalias in 384 ms
[13:40:52    21 s] tests service image: docker.io/some/image:tag healthcheck failed
[13:40:52    21 s] tests   Command failed with exit code 125: docker run --rm --name=gcl-wait-for-it-967338-0-4444 --network gitlab-ci-local-967338 willwill/wait-for-it myalias:4444 -t 30
[13:40:52    21 s] tests   Error: short-name resolution enforced but cannot prompt without a TTY
[...]
```

Reply:
Good point, some `podman` setups don't have `docker.io` as default registry.

In fact it would make sense to add `docker.io/` to all image references in the repository, including examples and test cases, to make it explicit that all images are to be found in the Docker Hub.

I'll prepare extra changes for that.